src/Temporalio/Worker/Tuning/SlotReserveContext.cs

```diff
@@ -0,0 +1,57 @@
+using Temporalio.Bridge;
+
+namespace Temporalio.Worker.Tuning
+{
+    /// <summary>
+    /// Context for reserving a slot from a <see cref="ICustomSlotSupplier"/>.
+    /// </summary>
+    /// <remarks>
+    /// WARNING: Custom slot suppliers are currently experimental.
+    /// </remarks>
+    public class SlotReserveContext
```

Comment:
I think this class and the other context/info classes should be `record`s with public fields and the logic for conversion from a bridge context to this remain with the bridge code. We should allow people to create these things themselves if they want to test their implementations.